### ISSUE TITLE
Missing [file] on line 355 in attachments controller

### ISSUE THREAD

**Issue by ViktorPontinen** [WP-API/WP-API#1303]
Hello,

Not sure if this is intended. Was having issues uploading in media endpoint and found that the code on line 355 was looking like this:
```
$file = wp_handle_upload( $files, $overrides );
``` 

Changed it to:
```
$file = wp_handle_upload( $files['file'], $overrides );
```

Which fixed uploading issues for me(using form-data).

**Comment by danielbachhuber** [comment]
Hm, it works for me locally in testing and e24e1f1299f38849f21c8f9099a1aad87437a8ad was TDD to fix the exact opposite of your change.

Can you share your sample request?

**Comment by ViktorPontinen** [comment]
Using version 2.0-beta1, perhaps it has been fixed after that.
The request:
```
POST /wp-json/wp/v2/media HTTP/1.1
Host: example.com
X-Admin-Public: xxxxxxx
X-Admin-Hash: xxxxxxxx
Cache-Control: no-cache
Postman-Token: xxxxxxx
Content-Type: multipart/form-data; boundary=----WebKitFormBoundary7MA4YWxkTrZu0gW

----WebKitFormBoundary7MA4YWxkTrZu0gW
Content-Disposition: form-data; name="file"; filename="avatar.gif"
Content-Type: image/gif


----WebKitFormBoundary7MA4YWxkTrZu0gW
```

Which gave the following error:
```
File is empty. Please upload something more substantial. This error could also be caused by uploads being disabled in your php.ini or by post_max_size being defined as smaller than upload_max_filesize in php.ini.
```
The php.ini was correctly configured since the attachment upload worked without problems on the  1.2.2 version.

Also have another issue on the same note, I have to send both "post" and "post_parent" fields, with the same value for both, for the attachment to get attached to a parent post, none of them works separately.

**Comment by rmccue** [comment]
This looks like a real bug to me, I think we have an issue with our test cases if it's not being picked up.

**Comment by kosso** [comment]
Uploading a media file was not working for me. After a lot of debugging, and following where the file was being tested, I found that as @ViktorPontinen also suggested, wp_handle_upload needs to have $files['file'], and not just $file.

`$file = wp_handle_upload( $files['file'], $overrides );`

**Comment by picklepete** [comment]
Just experienced this too, as @kosso said, the change to...

```php
$file = wp_handle_upload( $files['file'], $overrides );
```

... [on this line](https://github.com/WP-API/WP-API/blob/33cc8a60c467f97336bb1e9b80ec8abf929f37e1/lib/endpoints/class-wp-rest-attachments-controller.php#L367) makes multipart/form-data uploads work.

**Comment by rmccue** [comment]
I've also heard this from others, 99% certain our tests are wrong here.

**Comment by BE-Webdesign** [comment]
@rmccue , @danielbachhuber , @rachelbaker , @joehoyle I believe this was solved and #1492 solves this specific issue.  Media uploads seem great to me through the API, so I don't think this is relevant anymore. The API supports both raw data and multipart-form data.  Let me know if I am wrong!

**Comment by danielbachhuber** [comment]
:hammer: :hammer: :hammer:
